### ISSUE TITLE
`allow_skip: false` allows next section on empty console

### ISSUE THREAD

**Issue by gaelso** [rstudio/learnr#616]
### Issue 

Current behavior of learnr exercises with YAML `progressive: true` and `allow_skip: false` doesn't condition the access to a next section based on the code submitted.

When clicking Continue / Next topic before any interaction with the exercise, the following message appears:

> You must complete the exercise in this section before continuing.

This is a bit misleading as clicking on the **Run code**  button on any code or even on an empty console unlocks access to the next section. 

I understand conditioning button behavior on code run in exercises chunks might be tricky, but it should at least check that users didn't just click **Run code** on an empty console.

## Steps to reproduce

Initiate a basic learnr tutorial (with Rstudio: File > New file > R Markdown... > from Template > interactive tutorial).

Replace YAML:

```
---
output: learnr::tutorial
---
```

With

```
---
output: 
  learnr::tutorial:
    progressive: true
    allow_skip: false
---
```

Click **Run code** on the first exercise without typing any code in to mark the exercise as completed and unlock access to the next section.

## Session info
learnr 0.10.1.9017 on Win10 machine

**Comment by gadenbuie** [comment]
Notes to myself as I look into this.

Whether or not a user can move forward in a section is determined by the logic in the continue button click event handler

https://github.com/rstudio/learnr/blob/9bc2a378438cfa10170738b3590b15383b0d45a4/inst/rmarkdown/templates/tutorial/resources/tutorial-format.js#L181-L192

The section completed state seems to originate from the call to `$fireSectionCompleted()` in `$fireProgressEvent()`, where it appears that any `exercise_submission` event sets `progressEvent.completed = true` for the exercise in question.

https://github.com/rstudio/learnr/blob/9bc2a378438cfa10170738b3590b15383b0d45a4/inst/lib/tutorial/tutorial.js#L257-L259

Unfortunately, at this point, we don't know anything about the user's submission. In theory, the `data` given to this method has a `label` and `correct` flag —

```json
{
  "label": "two-plus-two",
  "correct": true
}
```

— but the `correct` flag is `true` by default when the code isn't checked, as in above (why not `null` or `undefined`??)

https://github.com/rstudio/learnr/blob/9bc2a378438cfa10170738b3590b15383b0d45a4/R/events_default.R#L139-L142

So it seems the lifecycle here is 

1. User click run button
2. Shiny receives data, broadcasts an `exercise_submission` event that corresponds to an `exercise_result` event (since `exercise_submission` event was renamed in #337 but was apparently kept around in the JS)
3. The progress event handler (`$fireProgressEvent()`) sets the exercise as completed because we got the `exercise_submission` event
4. The section unlocks because the exercise is complete.

After writing this out, I'm going to start by testing:

1. Only set `correct` when submission doesn't require a check
2. Add `completed` to the data that is `TRUE` when there is submission code and
    - the exercise doesn't have any grading
    - OR the exercise has grading code and the exercise was checked
3. Update the progress event handling to use this `completed` flag to set as completed if not yet complete

**Comment by gadenbuie** [comment]
Fixed by #633
